File: src/main/java/sharev/domain/team/repository/TeamRepositoryCustom.java
package sharev.domain.team.repository;

import java.util.List;
import sharev.domain.team.dto.response.ResponseTeamInfoDto;
import sharev.domain.team.dto.response.TempTeamMemberInfoDto;

public interface TeamRepositoryCustom {

    List<ResponseTeamInfoDto> findMyTeams(Long accountId);

    List<TempTeamMemberInfoDto> findMyTeamMembers(Long teamId);
}

File: src/main/kotlin/sharev/team/adapter/outbound/jpa/repository/TeamRepositoryCustom.kt
package sharev.team.adapter.outbound.jpa.repository

import sharev.team.application.port.outbound.summary.MyTeamSummary
import sharev.team.application.port.outbound.summary.TeamMemberSummary

interface TeamRepositoryCustom {
    fun findMyTeams(accountId: Long): List<MyTeamSummary>

    fun findMyTeamMembers(teamId: Long): List<TeamMemberSummary>
}
